arch/X86/X86NasmInstPrinter.c

```diff
@@ -0,0 +1,844 @@
+static void printRegName(SStream *OS , unsigned RegNo) {
+	SStream_concat0(OS , getRegisterName(RegNo));
+}
+
+// local printOperand, without updating public operands
+static void _printOperand(MCInst *MI , unsigned OpNo , SStream *O) {
+	MCOperand *Op  = MCInst_getOperand(MI , OpNo);
+	if (MCOperand_isReg(Op)) {
+		printRegName(O , MCOperand_getReg(Op));
+	}
+	else if (MCOperand_isImm(Op)) {
+		int64_t imm = MCOperand_getImm(Op);
+		if (imm < 0) {
+			if (imm < -HEX_THRESHOLD)
+				SStream_concat(O , "-0%"PRIx64"h" , -imm);
+			else
+				SStream_concat(O , "-%"PRIu64 , -imm);
+
+		}
+		else {
+			if (imm > HEX_THRESHOLD)
+				SStream_concat(O , "0%"PRIx64"h" , imm);
+			else
+				SStream_concat(O , "%"PRIu64 , imm);
+		}
+	}
+}
+
+static void printSrcIdx(MCInst *MI , unsigned Op , SStream *O) {
+	MCOperand *SegReg;
+	int reg;
+
+	if (MI->csh->detail) {
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].type = X86_OP_MEM;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].size = MI->x86opsize;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.segment = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.base = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.index = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.scale = 1;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.disp = 0;
+	}
+
+	SegReg = MCInst_getOperand(MI , Op + 1);
+	reg = MCOperand_getReg(SegReg);
+
+	// If this has a segment register, print it.
+	if (reg) {
+		_printOperand(MI , Op + 1 , O);
+		if (MI->csh->detail) {
+			MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.segment = reg;
+		}
+		SStream_concat0(O , ":");
+	}
+
+	SStream_concat0(O , "[");
+	set_mem_access(MI , true);
+	printOperand(MI , Op , O);
+	SStream_concat0(O , "]");
+	set_mem_access(MI , false);
+}
+
+static void printDstIdx(MCInst *MI , unsigned Op , SStream *O) {
+	if (MI->csh->detail) {
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].type = X86_OP_MEM;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].size = MI->x86opsize;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.segment = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.base = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.index = X86_REG_INVALID;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.scale = 1;
+		MI->flat_insn->detail->x86.operands[MI->flat_insn->detail->x86.op_count].mem.disp = 0;
+	}
+
+	// DI accesses are always ES-based on non-64bit mode
+	if (MI->csh->mode != CS_MODE_64) {
+		SStream_concat(O , "es:[");
```

Comment:
This is not NASM syntax. Just try using `ndisasm` on bytes `2e 8b 03`, and you'll get `mov eax,[cs:ebx]`, not `mov eax,cs:[ebx]`. The former can't even be assembled by NASM.